state/keys_test.go

```diff
@@ -181,3 +182,57 @@ func TestKeysMarshalingFuzz(t *testing.T) {
 		require.True(keys.compare(decodedKeys))
 	}
 }
+
+func TestNewPermissionFromString(t *testing.T) {
+	require := require.New(t)
+	tests := []struct {
+		strPerm string
+		perm    Permissions
+	}{
+		{
+			strPerm: "read",
+			perm:    Read,
+		},
+		{
+			strPerm: "write",
+			perm:    Write,
+		},
+		{
+			strPerm: "allocate",
+			perm:    Allocate,
+		},
+		{
+			strPerm: "all",
+			perm:    All,
+		},
+		{
+			strPerm: "none",
+			perm:    None,
+		},
+	}
+
+	for i, test := range tests {
+		t.Run(fmt.Sprintf("%d", i), func(t *testing.T) {
+			var perm Permissions
+			err := perm.UnmarshalText([]byte(test.strPerm))
+			require.NoError(err)
+			require.Equal(test.perm, perm)
+		})
+	}
+
+	t.Run("InvalidString", func(t *testing.T) {
+		var perm Permissions
+		err := perm.UnmarshalText([]byte{byte(1), byte(10)})
+		require.ErrorIs(err, errInvalidHexadecimalString)
+	})
```

Comment:
could we update the struct definition to include `expectedErr` and move this test case alongside the others rather than breaking it out?

Reply:
done 👍